configlib/config.go

```diff
@@ -66,3 +69,92 @@ func IsCustomizationSet(key string, elems []interface{}, elem string) bool {
 	}
 	return false
 }
+
+// AddPackage add a package to the Package section of the yml config file
+func AddPackage(name string) error {
+	cfgname := viper.ConfigFileUsed()
+	err := add(cfgname, name)
+	if err != nil {
+		return err
+	}
+	err = LoadConfigFromPath(cfgname)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// add add a package to the Package section of the yml config file
+func add(ymlfile string, packageName string) error {
+	appFS := afero.NewOsFs()
+	fi, _ := os.Stat(ymlfile)
+	yf, err := afero.ReadFile(appFS, ymlfile)
+	if err != nil {
+		return err
+	}
+
+	if bytes.Contains(yf, []byte(packageName)) {
+		log.Info(fmt.Sprintf("Package <%s> already found in <%s>", packageName, ymlfile))
+		return nil
+	}
+
+	var out []byte
+	i := 0
+	lines := bytes.Split(yf, []byte("\n"))
+	for _, line := range lines {
+		i++
+		out = append(out, line...)
+		if i < len(lines) {
+			out = append(out, []byte("\n")...)
+		}
+		if bytes.HasPrefix(line, []byte("Packages:")) {
+			out = append(out, []byte("  - "+packageName)...)
```

Comment:
why not add the newline after pkgname in a single []bytes, instead of twice?

Reply:
It's all about not writing the newline for the last line. The NL is written if it's not the last line of the file, or if we are adding the package.
One important unit test is to add a package, then remove the same package, and then have the same exact file (without ignoring whitespace)